src/main/java/life/qbic/omero/BasicOMEROClient.java

```diff
@@ -143,13 +155,25 @@ private void connect(String username, String password, String hostname, int port
 
     try {
       ExperimenterData user = this.gateway.connect(loginCredentials);
-      this.securityContext = new SecurityContext(user.getGroupId());
+
+      long userGroupID = user.getGroupId();
+
+      if(!group.equalsIgnoreCase("default")){
+        for (int i = 0; i < user.getGroups().size(); i++) {
+          if(user.getGroups().get(i).getName().equalsIgnoreCase(group)){
+            userGroupID = user.getGroups().get(i).getGroupId();
+            break;
```

Comment:
```suggestion
        for (GroupData userGroup : user.getGroups()) {
          if(userGroup.getName().equalsIgnoreCase(group)){
            userGroupID = userGroup.getGroupId();
            break;
```

I think this can at least be made clearer without using streams. You will need to import omero.gateway.model.GroupData as well.